src/http_server.cc

```diff
@@ -243,8 +243,6 @@ TRITONSERVER_Error*
 JsonBytesArrayByteSize(
     triton::common::TritonJson::Value& tensor_data, size_t* byte_size)
 {
-  *byte_size = 0;
```

Comment:
Actual I would prefer to keep the function like this and accumulate byte_size for elements in a separate variable in L254

Reply:
In the end you can set byte_size to the accumulated byte size.

Reply:
How about rename the recursive function to `FindJsonBytesArrayByteSizeHelper`, and wrap it with `FindJsonBytesArrayByteSize` that will initialize `*byte_size = 0;`?

LIke this:
```
TRITONSERVER_Error*
FindJsonBytesArrayByteSizeHelper(
    triton::common::TritonJson::Value& tensor_data, size_t* byte_size)
{
  for (size_t i = 0; i < tensor_data.ArraySize(); i++) {
    triton::common::TritonJson::Value el;
    RETURN_IF_ERR(tensor_data.At(i, &el));

    // Recurse if not last dimension...
    TRITONSERVER_Error* assert_err =
        el.AssertType(triton::common::TritonJson::ValueType::ARRAY);
    if (assert_err == nullptr) {
      RETURN_IF_ERR(JsonBytesArrayByteSize(el, byte_size));
    } else {
      // Serialized data size is the length of the string itself plus
      // 4 bytes to record the string length.
      const char* str;
      size_t len = 0;
      RETURN_MSG_IF_ERR(
          el.AsString(&str, &len), "Unable to parse JSON bytes array");
      *byte_size += len + sizeof(uint32_t);
    }
    TRITONSERVER_ErrorDelete(assert_err);
  }

  return nullptr;  // success
}

TRITONSERVER_Error*
FindJsonBytesArrayByteSize(
    triton::common::TritonJson::Value& tensor_data, size_t* byte_size)
{
  *byte_size = 0;
  return FindJsonBytesArrayByteSizeHelper(tensor_data, byte_size);
}

... skip lines ...
          if (dtype == TRITONSERVER_TYPE_BYTES) {
            RETURN_IF_ERR(FindJsonBytesArrayByteSize(tensor_data, &byte_size));
          } else {
            byte_size = element_cnt * TRITONSERVER_DataTypeByteSize(dtype);
          }
```

Reply:
I see. Do you mean this? 
```
--- a/src/http_server.cc
+++ b/src/http_server.cc
@@ -243,6 +243,8 @@ TRITONSERVER_Error*
 JsonBytesArrayByteSize(
     triton::common::TritonJson::Value& tensor_data, size_t* byte_size)
 {
+  *byte_size = 0;
+
   for (size_t i = 0; i < tensor_data.ArraySize(); i++) {
     triton::common::TritonJson::Value el;
     RETURN_IF_ERR(tensor_data.At(i, &el));
@@ -251,7 +253,9 @@ JsonBytesArrayByteSize(
     TRITONSERVER_Error* assert_err =
         el.AssertType(triton::common::TritonJson::ValueType::ARRAY);
     if (assert_err == nullptr) {
-      RETURN_IF_ERR(JsonBytesArrayByteSize(el, byte_size));
+      size_t byte_size_;
+      RETURN_IF_ERR(JsonBytesArrayByteSize(el, &byte_size_));
+      *byte_size += byte_size_;
     } else {
       // Serialized data size is the length of the string itself plus
       // 4 bytes to record the string length.
@@ -2419,7 +2423,6 @@ HTTPAPIServer::EVBufferToInput(
               "Unable to parse 'data'");
 
           if (dtype == TRITONSERVER_TYPE_BYTES) {
-            byte_size = 0;  // initialize for its recursive computation
             RETURN_IF_ERR(JsonBytesArrayByteSize(tensor_data, &byte_size));
           } else {
             byte_size = element_cnt * TRITONSERVER_DataTypeByteSize(dtype);
```

Reply:
Yes. Something like this.

Reply:
Ok, updated.